### PR TITLE
Fix the tests don't run correctly on CI

### PR DESCRIPTION
**Pull request by watilde**
The tests doesn't work on local and CI as well. This PR is to just run tests without fixing actual test cases.

### CODE REVIEW

**Comment by watilde** [comment]
@andyearnshaw Hi. I made ci tests run completely and I'd like to keep this PR  as fix only the thing if it's ok. Could you please take a look into this when you have a time? Here are the results of ci tests:

Master branch
+ v0.10.x: https://travis-ci.org/andyearnshaw/Intl.js/jobs/125975472
+ v0.12.x: https://travis-ci.org/andyearnshaw/Intl.js/jobs/125975473

This patch
+ v0.10.x: https://travis-ci.org/andyearnshaw/Intl.js/jobs/126108727
+ v0.12.x: https://travis-ci.org/andyearnshaw/Intl.js/jobs/126108728

File: Gruntfile.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ module.exports = function (grunt) {
                 dest  : 'tests/test262',
                 src   : [
                     'LICENSE',
-                    'test/intl402/*.js',
+                    'test/intl402/**/*.js',
```
Comment (watilde) [inline Gruntfile.js]: Added some directories in [test262/test/intl402](https://github.com/tc39/test262/tree/master/test/intl402) so this pattern should be `**/*.js`.

File: src/main.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import IntlPolyfill from "./core.js";
 
 // hack to export the polyfill as global Intl if needed
-if (typeof Intl !== undefined) {
-    Intl = IntlPolyfill;
-    IntlPolyfill.__applyLocaleSensitivePrototypes();
+if (typeof Intl !== 'undefined') {
+    try {
```
Comment (caridy) [inline src/main.js]: I'm not sure about this try/catch, and the silence failure when trying to apply sensitive prototypes.

**Comment by caridy** [comment]
@watilde thanks a lot for the help, I will be merging this later today.

File: src/main.js (hunk 2 above)
Comment (watilde) [inline src/main.js]: Honesty I'm not sure about it too. It happened on the following CI test and I couldn't reproduce it on my local in spite of using same version of `Node.js`/`npm`, but I could see `Cannot assign to read only property`. That's why I added it.
https://travis-ci.org/andyearnshaw/Intl.js/jobs/126101600

**Comment by caridy** [comment]
@watilde I went on and fixes the whole test harness to be able to run the latest test, locally, I'm getting 99 passing correctly, no failures, that's a huge improvements :), thanks a lot again.

On travis I'm getting few failures:

* "passCount": 553,
* "failCount": 18,

But this is a huge step in the right direction.

**Comment by watilde** [comment]
@caridy Wow I'm glad to hear that! I was trying to fix failed errors too, for latest tests on my local.

As a first point I started fixing `6.4 Timezone Names` but finally I couldn't fix it. There was now way to get user's current time zone correctly because it's implemented in platform codes(e.g. [win32](https://github.com/v8/v8/blob/47cce544a31ed5577ffe2963f67acb4144ee0232/src/base/platform/platform-win32.cc), [macos](https://github.com/v8/v8/blob/47cce544a31ed5577ffe2963f67acb4144ee0232/src/base/platform/platform-macos.cc)) and there is no interface without Intl native Object currently. Hence, I knew that's was it was mentioned as `Not Implemented` in the `README.md` of this project. Well, I'm happy to support fix tests😃

**Pull request merged**